{"schemaVersion":{"patch":0,"major":0,"minor":3},"hierarchy":{"paths":[["doc:\/\/AsyncObjects\/documentation\/AsyncObjects","doc:\/\/AsyncObjects\/documentation\/AsyncObjects\/Continuable"]]},"variants":[{"traits":[{"interfaceLanguage":"swift"}],"paths":["\/documentation\/asyncobjects\/continuable\/resume(with:)"]}],"metadata":{"title":"resume(with:)","required":true,"modules":[{"name":"AsyncObjects"}],"externalID":"s:12AsyncObjects11ContinuableP6resume4withys6ResultOy7SuccessQz7FailureQzG_tF","symbolKind":"method","fragments":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"text":"resume","kind":"identifier"},{"text":"(","kind":"text"},{"text":"with","kind":"externalParam"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:s6ResultO","text":"Result","kind":"typeIdentifier"},{"text":"<","kind":"text"},{"text":"Self","kind":"typeIdentifier"},{"text":".","kind":"text"},{"text":"Success","kind":"typeIdentifier"},{"text":", ","kind":"text"},{"text":"Self","kind":"typeIdentifier"},{"text":".","kind":"text"},{"text":"Failure","kind":"typeIdentifier"},{"kind":"text","text":">)"}],"roleHeading":"Instance Method","role":"symbol"},"identifier":{"url":"doc:\/\/AsyncObjects\/documentation\/AsyncObjects\/Continuable\/resume(with:)","interfaceLanguage":"swift"},"sections":[],"kind":"symbol","abstract":[{"text":"Resume the task awaiting the continuation by having it either return normally","type":"text"},{"type":"text","text":" "},{"type":"text","text":"or throw an error based on the state of the given "},{"code":"Result","type":"codeVoice"},{"text":" value.","type":"text"}],"primaryContentSections":[{"kind":"declarations","declarations":[{"platforms":["Linux"],"tokens":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"text":"resume","kind":"identifier"},{"text":"(","kind":"text"},{"text":"with","kind":"externalParam"},{"text":" ","kind":"text"},{"text":"result","kind":"internalParam"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:s6ResultO","text":"Result","kind":"typeIdentifier"},{"text":"<","kind":"text"},{"text":"Self","kind":"typeIdentifier"},{"text":".","kind":"text"},{"text":"Success","kind":"typeIdentifier"},{"text":", ","kind":"text"},{"text":"Self","kind":"typeIdentifier"},{"text":".","kind":"text"},{"text":"Failure","kind":"typeIdentifier"},{"text":">)","kind":"text"}],"languages":["swift"]}]},{"kind":"parameters","parameters":[{"content":[{"type":"paragraph","inlineContent":[{"text":"A value to either return or throw from the continuation.","type":"text"}]}],"name":"result"}]}],"references":{"doc://AsyncObjects/documentation/AsyncObjects/Continuable":{"title":"Continuable","url":"\/documentation\/asyncobjects\/continuable","type":"topic","fragments":[{"text":"protocol","kind":"keyword"},{"text":" ","kind":"text"},{"text":"Continuable","kind":"identifier"}],"role":"symbol","kind":"symbol","abstract":[{"text":"A type that allows to interface between synchronous and asynchronous code,","type":"text"},{"text":" ","type":"text"},{"text":"by representing task state and allowing task resuming with some value or error.","type":"text"}],"identifier":"doc:\/\/AsyncObjects\/documentation\/AsyncObjects\/Continuable","navigatorTitle":[{"text":"Continuable","kind":"identifier"}]},"doc://AsyncObjects/documentation/AsyncObjects/Continuable/resume(with:)":{"identifier":"doc:\/\/AsyncObjects\/documentation\/AsyncObjects\/Continuable\/resume(with:)","title":"resume(with:)","required":true,"url":"\/documentation\/asyncobjects\/continuable\/resume(with:)","fragments":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"text":"resume","kind":"identifier"},{"text":"(","kind":"text"},{"text":"with","kind":"externalParam"},{"text":": ","kind":"text"},{"text":"Result","preciseIdentifier":"s:s6ResultO","kind":"typeIdentifier"},{"text":"<","kind":"text"},{"text":"Self","kind":"typeIdentifier"},{"text":".","kind":"text"},{"text":"Success","kind":"typeIdentifier"},{"kind":"text","text":", "},{"kind":"typeIdentifier","text":"Self"},{"kind":"text","text":"."},{"text":"Failure","kind":"typeIdentifier"},{"text":">)","kind":"text"}],"kind":"symbol","type":"topic","role":"symbol","abstract":[{"text":"Resume the task awaiting the continuation by having it either return normally","type":"text"},{"text":" ","type":"text"},{"text":"or throw an error based on the state of the given ","type":"text"},{"code":"Result","type":"codeVoice"},{"text":" value.","type":"text"}]},"doc://AsyncObjects/documentation/AsyncObjects":{"abstract":[{"text":"Several synchronization primitives and task synchronization mechanisms introduced to aid in modern swift concurrency.","type":"text"}],"role":"collection","title":"AsyncObjects","url":"\/documentation\/asyncobjects","identifier":"doc:\/\/AsyncObjects\/documentation\/AsyncObjects","kind":"symbol","type":"topic"}}}